target_mailchimp_v2/sinks.py

```diff
@@ -389,44 +403,53 @@ def preprocess_record(self, record: dict, context: dict) -> None:
-            # custom logic for contacts
-            if self.stream_name.lower() in self.contact_names:
-                # add email to the endpoint to use create or update endpoint
-                email = record.get("email_address")
-                if not email:
-                    raise Exception(
-                        f"No email found for record {record}, email is a required field."
-                    )
-                # get list id
-                list_id = self.get_list_id()
-                if not list_id:
-                    raise Exception(
-                        f"No list id found to send record with email {email}"
-                    )
-                # using create or update endpoint for contacts
-                method = "PUT"
-                endpoint = f"/lists/{list_id}/members/{email}"
-
-            # send data
-            id = record.pop("id", None)
-            if id:
-                id = int(id)
-                method = "PUT"
-                endpoint = f"{endpoint}/{id}"
-
-            # send data
-            client = MailchimpMarketing.ApiClient()
-            client.set_config(
-                {
-                    "access_token": self.config.get("access_token"),
-                    "server": self.get_server(),
-                }
-            )
-            response = client.call_api(
-                resource_path=endpoint, method=method, body=record
+        client_api = self.get_client_API()
+        client = self.get_client()
+        id = record.pop("id", None)
+        
+        # custom logic for contacts
+        if self.stream_name.lower() in self.contact_names:
+            # add email to the endpoint to use create or update endpoint
+            email = record.get("email_address")
+            if not email:
+                raise Exception(
+                    f"No email found for record {record}, email is a required field."
+                )
+            # get list id
+            list_id = self.get_list_id()
+            if not list_id:
+                raise Exception(
+                    f"No list id found to send record with email {email} for list {self.config.get('list_name')}"
+                )
+            # using create or update endpoint for contacts
+            method = "PUT"
+            endpoint = f"/lists/{list_id}/members/{email}"
+            self.get_merge_fields(record, list_id)
+        elif id:
+            endpoint = f"{endpoint}/{id}"
+
+        json_serializable_record = self.clean_convert(record)
+        try:
+            response = client_api.call_api(
+                resource_path=endpoint, method=method, body=json_serializable_record
             )
-            id = response["id"]
-            return id, True, state_updates
\ No newline at end of file
+            id = response["id"] if "id" in response else id if id else ""
+            return id, True, state_updates
+        except Exception as e:
+            return "", False, state_updates
```

Comment:
I think here you could just raise the exception, in that way the state would add it automatically to the state so we can see the cause of failure

Reply:
Fixed! Thanks 🙏